common/error.h

```diff
@@ -50,6 +50,13 @@ class [[nodiscard]] Error {
     out << message();
   }
 
+  // Return another error with the prefix replaced by `prefix`.
+  auto WithPrefix(llvm::Twine prefix) && -> Error {
```

Comment:
I'm less convinced with these changes that `prefix` is needed anymore (particularly since you're discarding any old prefix). WDYT if we removed the prefix here? The code implementing prefixes could be more local to the code adding them right now.

Or, at least, this should probably require that it's not removing a prefix.

Reply:
Nice! Removed `prefix`.